ast-nodes/for-loop.dart

```diff
@@ -76,8 +76,12 @@ class ForLoop implements Statement, ScopeCreator {
 
     var scope = Scope();
     this.scopes = [scope];
-    ScopeElement currentMark = scope.addDeclaration(
-        VariableDeclaration(this.loopVariable.name, IntegerType(), null));
+    
+    var declaration = VariableDeclaration(this.loopVariable.name, IntegerType(), null);
+    declaration.scopeMark = parentMark;
+    declaration.scopeMark.readOnly = true;
```

Comment:
Wait, no, the scope mark of this declaration will not be this declaration. You need to set this on `currentMark`

Reply:
But it worked, and then I change readOnly of currentMark, it doesn't work